doc/reference/javascript_reference.rst

```diff
@@ -0,0 +1,1806 @@
+        <text xml:space="preserve" text-anchor="start" font-family="Helvetica, Arial, sans-serif" font-size="24" id="svg_12" y="128.5" x="276" fill-opacity="null" stroke-opacity="null" stroke-width="0" stroke="#000" fill="#000000">Renderer</text>
+        <text xml:space="preserve" text-anchor="start" font-family="Helvetica, Arial, sans-serif" font-size="24" id="svg_13" y="127.5" x="442" fill-opacity="null" stroke-opacity="null" stroke-width="0" stroke="#000" fill="#000000">Model</text>
+    </svg>
+
+- the View is the factory. Its job is to get a set of fields, arch, context and
+  some other parameters, then to construct a Controller/Renderer/Model triplet.
+
+  The view's role is to properly setup each other piece, with the correct
+  information.  Usually, it has to process the arch string and extract the
+  data necessary for each other parts of the view.
+
+  Note that the view is a class, not a widget.  Once its job has been done, it
+  can be discarded.
+
+- the Renderer has one job: represent the data being viewed in a DOM element.
+  Each view can render the data in a different way.  Also, it should listen on
+  appropriate user actions and notify its parent (the Controller) if necessary.
+
+  The Renderer is the V in the MVC pattern.
+
+- the Model: its job is to fetch and hold the state of the view.  Usually, it
+  represents in some way a set of records in the database.  The Model is the
+  owner of the 'business data'. It is the M in the MVC pattern.
+
+- the Controller: its job is to coordinate the renderer and the model.  Also, it
+  is the main entry point for the rest of the web client.  For example, when
+  the user change something in the search view, the *update* method of the
+  controller will be called with the appropriate information.
+
+  It is the C in the MVC pattern.
+
+Note: the JS code for the views has been designed to be usable outside of the
+context of a view manager/action manager.  They could be used in a client action,
+or, they could be displayed in the public website (with some work on the assets).
+
+Field Widgets
+=============
+
+A good part of the web client experience is about editing and creating data. Most
+of that work is done with the help of widgets, which are aware of the field
+type and of the specific details on how a value should be displayed and edited.
+
+AbstractField
+-------------
+
+The *AbstractField* class is the base class for all widgets in a view, for all
+views that support them (currently: Form, List, Kanban).
+
+There are many differences between the v11 field widgets and the previous versions.
+Let us mention the most important ones:
+
+- the widgets are shared between all views (well, Form/List/Kanban). No need to
+  duplicate the implementation anymore.
```

Comment:
maybe mention that if one wants a specific implementation in a view type, it can be done by prefixing the key by the view type in the field widgets registry